web-ng/src/app/components/drawing-tools/drawing-tools.component.spec.ts

```diff
@@ -0,0 +1,324 @@
+/**
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  ComponentFixture,
+  fakeAsync,
+  flush,
+  TestBed,
+  tick,
+  waitForAsync,
+} from '@angular/core/testing';
+import { BehaviorSubject, of } from 'rxjs';
+import {
+  DrawingToolsService,
+  EditMode,
+} from '../../services/drawing-tools/drawing-tools.service';
+import { NavigationService } from '../../services/navigation/navigation.service';
+import { ProjectService } from '../../services/project/project.service';
+import { Layer } from '../../shared/models/layer.model';
+import { Project } from '../../shared/models/project.model';
+import { StringMap } from '../../shared/models/string-map.model';
+import { DrawingToolsComponent } from './drawing-tools.component';
+import { DrawingToolsModule } from './drawing-tools.module';
+import { Map } from 'immutable';
+import { AuthService } from '../../services/auth/auth.service';
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+import { urlPrefix } from '../map/ground-pin';
+import { By } from '@angular/platform-browser';
+
+describe('DrawingToolsComponent', () => {
+  let fixture: ComponentFixture<DrawingToolsComponent>;
+
+  let authServiceSpy: jasmine.SpyObj<AuthService>;
+  let mockDisabled$: BehaviorSubject<boolean>;
+  let mockEditMode$: BehaviorSubject<EditMode>;
+  let drawingToolsServiceSpy: jasmine.SpyObj<DrawingToolsService>;
+  let mockObservationId$: BehaviorSubject<string | null>;
+  let navigationServiceSpy: jasmine.SpyObj<NavigationService>;
+  let projectServiceSpy: jasmine.SpyObj<ProjectService>;
+
+  const layerId1 = 'layer001';
```

Comment:
There's a lot of test data here. Once I merge may you can use `TestData` builders as I'm proposing in #844?

Reply:
Sg, thanks for adding it.